File: app/src/main/java/com/example/orderingapp/Adapter/CategoryAdapter.java
package com.example.orderingapp.Adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.example.orderingapp.Domain.CategoryModel;
import com.example.orderingapp.R;
import com.example.orderingapp.databinding.ViewholderCategoryBinding;

import java.util.List;

public class CategoryAdapter extends RecyclerView.Adapter<CategoryAdapter.ViewHolder> {

    private Context context;
    private List<CategoryModel> items;
    private int selectedPosition = -1;
    private int lastSelectedPosition = -1;

    public CategoryAdapter(List<CategoryModel> items) {
        this.items = items;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        ViewholderCategoryBinding binding;

        public ViewHolder(ViewholderCategoryBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        context = parent.getContext();
        LayoutInflater inflater = LayoutInflater.from(context);
        ViewholderCategoryBinding binding = ViewholderCategoryBinding.inflate(inflater, parent, false);
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, @SuppressLint("RecyclerView") int position) {
        CategoryModel item = items.get(position);
        holder.binding.title.setText(item.getTitle());

        holder.binding.getRoot().setOnClickListener(v -> {
            lastSelectedPosition = selectedPosition;
            selectedPosition = position;
            if (lastSelectedPosition != -1) {
                notifyItemChanged(lastSelectedPosition);
            }
            notifyItemChanged(selectedPosition);
        });

        if (selectedPosition == position) {
            holder.binding.title.setBackgroundResource(R.drawable.greenish_bg);
            holder.binding.title.setTextColor(ContextCompat.getColor(context, R.color.Deepteal));
        } else {
            holder.binding.title.setBackgroundResource(R.drawable.coralpink_bg);
            holder.binding.title.setTextColor(ContextCompat.getColor(context, R.color.Deepteal));
        }
    }

    @Override
    public int getItemCount() {
        return items.size();
    }
}

File: app/src/main/java/com/example/orderingapp/Adapta/CategoryAdapter.kt
package com.example.orderingapp.Adapta

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.RemoteViews.RemoteCollectionItems
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.example.orderingapp.Domain.CategoryModel
import com.example.orderingapp.R
import com.example.orderingapp.databinding.ViewholderCategoryBinding

class CategoryAdapter(val items: MutableList<CategoryModel>) :
    RecyclerView.Adapter<CategoryAdapter.Viewholder>() {

    private lateinit var context: Context
    private var selectedPosition = -1
    private var lastselectedPosition = -1

    inner class Viewholder(val binding: ViewholderCategoryBinding)
        :RecyclerView.ViewHolder(binding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CategoryAdapter.Viewholder {
        context = parent.context
        val binding = ViewholderCategoryBinding.inflate(LayoutInflater.from(context), parent, false)
        return Viewholder(binding)
    }

    override fun onBindViewHolder(holder: CategoryAdapter.Viewholder, position: Int) {
        val item = items[position]
        holder.binding.title.text = item.title

        holder.binding.root.setOnClickListener {
            lastselectedPosition=selectedPosition
            selectedPosition = position
            notifyItemChanged(lastselectedPosition)
            notifyItemChanged(selectedPosition)
        }

        // Check if context is valid before using it
            if (selectedPosition == position) {
                holder.binding.title.setBackgroundResource(R.drawable.greenish_bg)
                holder.binding.title.setTextColor(ContextCompat.getColor(context, R.color.Deepteal))
            } else {
                holder.binding.title.setBackgroundResource(R.drawable.coralpink_bg)
                holder.binding.title.setTextColor(context.resources.getColor(R.color.Deepteal))
            }
    }

    override fun getItemCount(): Int = items.size
}
